File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

# [[2] [4 5]]
# [[3] [6 7]]

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        if not root:
            return []

        result = []

        def dfs(node, depth):
            if not node:
                return None
            
            if len(result) == depth:
                result.append([])
            
            result[depth].append(node.val)
            depth += 1
            dfs(node.left, depth)
            dfs(node.right, depth)

        dfs(root, 0)

        return result
            

File: Data Structures & Algorithms/binary-tree-preorder-traversal/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> preorderTraversal(TreeNode root) {
        if (root == null){
            return new ArrayList<Integer>();
        }
        List<Integer> result = new ArrayList<>();
        result.add(root.val);
        if (root.left != null){
            List<Integer> leftTree = preorderTraversal(root.left);
            result.addAll(leftTree);
        }
        if (root.right != null){
            List<Integer> rightTree = preorderTraversal(root.right);
            result.addAll(rightTree);
        }
        return result;
    }
}